test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs

```diff
@@ -7321,7 +7322,297 @@ public async Task TestPacManPreview_InstallForPC_PackageNamespace_WithMultipleFe
+            // Arrange
+            using (var testSolutionManager = new TestSolutionManager())
+            {
+                // Set up Package Source
+                var primarySources = new List<PackageSource>();
+                var secondarySources = new List<PackageSource>();
+                var externalRepositoryPath = Path.Combine(testSolutionManager.SolutionDirectory, "ExternalRepository");
+                Directory.CreateDirectory(externalRepositoryPath);
+
+                var privateRepositoryPath = Path.Combine(testSolutionManager.SolutionDirectory, "PrivateRepository");
+                Directory.CreateDirectory(privateRepositoryPath);
+
+                // Replace the default nuget.config with custom one.
+                SettingsTestUtils.CreateConfigurationFile(testSolutionManager.NuGetConfigPath, $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+    <clear />
+    <add key=""ExternalRepository"" value=""{externalRepositoryPath}"" />
+    <add key=""PrivateRepository"" value=""{privateRepositoryPath}"" />
+    </packageSources>
+    <packageNamespaces>
+        <packageSource key=""externalRepository"">
+            <namespace id=""External.*"" />
+            <namespace id=""Others.*"" />
+        </packageSource>
+        <packageSource key=""PrivateRepository"">
+            <namespace id=""Contoso.*"" />             
+            <namespace id=""Test.*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+                var contosoPackageIdentity = new PackageIdentity("Contoso.A", new NuGetVersion("1.0.0"));
+                var directPackageIdentity = new PackageIdentity("Direct.A", new NuGetVersion("1.0.0"));
+
+                var ExternalContosoA = new SimpleTestPackageContext()
+                {
+                    Id = contosoPackageIdentity.Id,  // Package id conflict with PrivateRepository
+                    Version = "1.0.0"
+                };
+                ExternalContosoA.AddFile("lib/net461/externalA.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    externalRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ExternalContosoA);
+
+                var ExternalDirectA = new SimpleTestPackageContext()
+                {
+                    Id = directPackageIdentity.Id, // Package Id not filtered by package namespace.
+                    Version = "1.0.0",
+                    Dependencies = new List<SimpleTestPackageContext>() { ExternalContosoA } // We set Contoso.A as dependent package.
+                };
+                ExternalDirectA.AddFile("lib/net461/directA.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
```

Comment:
you are mkaing a similar call for both external direct and externalcontoso.

Reply:
Ok. I didn't know we can create multiple nupkgs same time. I combined them into 1 call.